sphinx/testing/_matcher/options.py

```diff
@@ -0,0 +1,241 @@
+    the duplicated lines, i.e., ``['a', 'b', '', 'b'] -> ['a', 'b']``.
+    """
+
+    unique: bool
+    """Eliminate multiple occurrences of lines in the output.
+
+    The default value is ``False``.
+
+    This option is only applied at the very end of the transformation chain,
+    after empty and duplicated consecutive lines might have been eliminated.
+    """
+
+    delete: DeletePattern
+    r"""Prefixes or patterns to remove from the output lines.
+
+    The transformation is described for one or more :class:`str`
+    or :class:`~re.Pattern` objects as follows:
+
+    - Compile :class:`str` pattern into :class:`~re.Pattern` according
+      to the pattern :attr:`flavor` and remove prefixes matching those
+      patterns from the output lines.
+    - Replace substrings in the output lines matching one or more
+      patterns directly given as :class:`~re.Pattern` objects.
+
+    The process is repeated until no output lines starts by any
+    of the given strings or matches any of the given patterns.
+
+    This transformation is applied at the end of the transformation
+    chain, just before filtering the output lines are filtered with
+    the :attr:`ignore` predicate.
+
+    Example::
+
+        clean('abcdA\n1', delete='abcd') == ['A', '1']
+        clean('1234A\nxyzt', delete=r'\d+', flavor='re') == ['A', 'xyzt']
+    """
+
+    ignore: LinePredicate | None
+    """A predicate for filtering the output lines.
+
+    Lines that satisfy this predicate are not included in the output.
+
+    The default is ``None``, meaning that all lines are included.
+    """
+
+    flavor: Flavor
+    """Indicate how strings are matched against non-compiled patterns.
+
+    The allowed values for :attr:`flavor` are:
+
+    * ``'none'`` -- match lines using string equality (the default).
+    * ``'fnmatch'`` -- match lines using :mod:`fnmatch`-style patterns.
+    * ``'re'`` -- match lines using :mod:`re`-style patterns.
+
+    This option only affects non-compiled patterns (i.e., those given
+    as :class:`str` and not :class:`~re.Pattern` objects).
+    """
+
+
+@final
+class CompleteOptions(TypedDict):
+    """Same as :class:`Options` but as a total dictionary.
+
+    :meta private:
+    """
+
+    # Whenever a new option in :class:`Options` is added, do not
+    # forget to add it here and in :data:`DEFAULT_OPTIONS`.
+
+    color: bool
+    ctrl: bool
+
+    strip: StripChars
+    stripline: StripChars
+
+    keepends: bool
+    empty: bool
+    compress: bool
+    unique: bool
+
+    delete: DeletePattern
+    ignore: LinePredicate | None
+
+    flavor: Flavor
+
+
+DEFAULT_OPTIONS: Final[CompleteOptions] = CompleteOptions(
+    color=False,
+    ctrl=True,
+    strip=True,
+    stripline=False,
+    keepends=False,
+    empty=True,
+    compress=False,
+    unique=False,
+    delete=(),
+    ignore=None,
+    flavor='none',
+)
+"""The default (read-only) options values."""
+
+if TYPE_CHECKING:
+    _OptionsView = Union[Options, CompleteOptions]
```

Comment:
I'm putting it here to avoid have explicit forward references at the top of the file.